src/Views/ProxyPage.vala

```diff
@@ -63,10 +92,74 @@ namespace Network.Widgets {
             stack.visible_child = configuration_page;
         }
 
+#if USE_UBUNTU_SYSTEM_SERVICE
+        private void on_proxy_settings_changed () {
+            if (!systemwide_available || system_proxy_service == null) {
+                return;
+            }
+
+            if (Network.Plug.proxy_settings.get_enum ("mode") == GDesktop.ProxyMode.MANUAL) {
+                var http_settings = Network.Plug.http_settings;
+                var https_settings = Network.Plug.https_settings;
+                var ftp_settings = Network.Plug.ftp_settings;
+                var socks_settings = Network.Plug.socks_settings;
+                try {
+                    var host = http_settings.get_string ("host");
+                    var port = http_settings.get_int ("port");
+                    if (host != "" && port > 0) {
+                        system_proxy_service.set_proxy ("http", "http://%s:%d/".printf (host, port));
+                    } else {
+                        system_proxy_service.set_proxy ("http", "");
+                    }
+
+                    host = https_settings.get_string ("host");
+                    port = https_settings.get_int ("port");
+                    if (host != "" && port > 0) {
+                        system_proxy_service.set_proxy ("https", "https://%s:%d/".printf (host, port));
+                    } else {
+                        system_proxy_service.set_proxy ("https", "");
+                    }
+
+                    host = ftp_settings.get_string ("host");
+                    port = ftp_settings.get_int ("port");
+                    if (host != null && port > 0) {
+                        system_proxy_service.set_proxy ("ftp", "ftp://%s:%d/".printf (host, port));
+                    } else {
+                        system_proxy_service.set_proxy ("ftp", "");
+                    }
+
+                    host = socks_settings.get_string ("host");
+                    port = socks_settings.get_int ("port");
+                    if (host != null && port > 0) {
+                        system_proxy_service.set_proxy ("socks", "socks://%s:%d/".printf (host, port));
+                    } else {
+                        system_proxy_service.set_proxy ("socks", "");
+                    }
+
+                    system_proxy_service.set_no_proxy (string.joinv (",", Network.Plug.proxy_settings.get_strv ("ignore-hosts")));
+                } catch (Error e) {
+                    warning ("Error applying systemwide proxy settings: %s", e.message);
+                }
+            } else {
+                try {
+                    system_proxy_service.set_proxy ("http", "");
+                    system_proxy_service.set_proxy ("https", "");
+                    system_proxy_service.set_proxy ("ftp", "");
+                    system_proxy_service.set_proxy ("socks", "");
+                    system_proxy_service.set_no_proxy ("");
```

Comment:
All of these calls are synchronous which means they are blocking. Now I know they will mostly take almost no time to complete but I think it's worth to make these async instead so that there is no possibility of blocking the UI (in case of e.g DBus timeout).